File: data/multilang_comparison/program_julia/src/func43.jl
include("program_julia_head.jl")
function func43(vars::JArray_param, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   array19 = nothing
   if pCounter > 1
       array19 = vars.data[pCounter]
       pCounter -= 1
       array19.refC += 1
   else
       array19 = JArray(zeros(Int, 814), 814, 1, 19)
   end
   return array19;
end

File: data/multilang_comparison/program_cpp_vector/src/func43.cpp
#include "program_cpp_vector.hpp" 
Array* func43(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array19;
   if (pCounter > 0) {
      array19 = vars->data.at(--pCounter);
      array19->refC++;
      DEBUG_COPY(array19->id);
   } else {
      array19 = new Array();
      array19->size = 814;
      array19->refC = 1;
      array19->id = 19;
      array19->data = std::vector<unsigned int>(array19->size, 0);
      DEBUG_NEW(array19->id);
   }
   return array19;
}

